iclib/bno055.py

```diff
@@ -0,0 +1,270 @@
+    """Lower Byte of SW Revision ID"""
+    SW_REV_ID_MSB = 0x05
+    """Upper Byte of SW Revision ID"""
+    BL_REV_ID = 0x06
+    """Identifies the version of the bootloader in the microcontroller"""
+    PAGE_ID = 0x07
+    """Number of currently selected page"""
+    ACC_DATA_X_LSB = 0x08
+    """Lower byte of x-axis acceleration data"""
+    ACC_DATA_X_MSB = 0x09
+    """Upper byte of x-axis acceleration data"""
+    ACC_DATA_Y_LSB = 0x0A
+    """Lower byte of Y axis acceleration data"""
+    ACC_DATA_Y_MSB = 0x0B
+    """Upper byte of y-axis acceleration data"""
+    ACC_DATA_Z_LSB = 0x0C
+    """Lower byte of z-axis acceleration data"""
+    ACC_DATA_Z_MSB = 0x0D
+    """Upper byte of z-axis acceleration data"""
+    MAG_DATA_X_LSB = 0x0E
+    """Lower byte of x-axis magnetometer data"""
+    MAG_DATA_X_MSB = 0x0F
+    """Upper byte of x-axis magnetometer data"""
+    MAG_DATA_Y_LSB = 0x10
+    """Lower byte of y-axis magnetometer data"""
+    MAG_DATA_Y_MSB = 0x11
+    """Upper byte of y-axis magnetometer data"""
+    MAG_DATA_Z_LSB = 0x12
+    """Lower byte of z-axis magnetometer data"""
+    MAG_DATA_Z_MSB = 0x13
+    """Upper byte of z-axis magnetometer data"""
+    GYR_DATA_X_LSB = 0x14
+    """Lower byte of x-axis gyroscope data"""
+    GYR_DATA_X_MSB = 0x15
+    """Upper byte of x-axis gyroscope data"""
+    GYR_DATA_Y_LSB = 0x16
+    """Lower byte of y-axis gyroscope data"""
+    GYR_DATA_Y_MSB = 0x17
+    GYR_DATA_Z_LSB = 0x18
+    GYR_DATA_Z_MSB = 0x19
+    EUL_DATA_X_LSB = 0x1A
+    EUL_DATA_X_MSB = 0x1B
+    EUL_DATA_Y_LSB = 0x1C
+    EUL_DATA_Y_MSB = 0x1D
+    EUL_DATA_Z_LSB = 0x1E
+    EUL_DATA_Z_MSB = 0x1F
+    QUA_DATA_W_LSB = 0x20
+    QUA_DATA_W_MSB = 0x21
+    QUA_DATA_X_LSB = 0x22
+    QUA_DATA_X_MSB = 0x23
+    QUA_DATA_Y_LSB = 0x24
+    QUA_DATA_Y_MSB = 0x25
+    QUA_DATA_Z_LSB = 0x26
+    QUA_DATA_Z_MSB = 0x27
+    LIA_DATA_X_LSB = 0x28
+    LIA_DATA_X_MSB = 0x29
+    LIA_DATA_Y_LSB = 0x2A
+    LIA_DATA_Y_MSB = 0x2B
+    LIA_DATA_Z_LSB = 0x2C
+    LIA_DATA_Z_MSB = 0x2D
+    GRV_DATA_X_LSB = 0x2E
+    GRV_DATA_X_MSB = 0x2F
+    GRV_DATA_Y_LSB = 0x30
+    GRV_DATA_Y_MSB = 0x31
+    GRV_DATA_Z_LSB = 0x32
+    GRV_DATA_Z_MSB = 0x33
+    TEMP = 0x34
+    CALIB_STAT = 0x35
+    UNIT_SEL = 0x3B
+    OPR_MODE = 0x3D
+
+    BNO055_ADDR = 0x28
+
+
+@dataclass
+class BNO055:
+
+    i2c: I2C
+    gpio_out_imu_reset: GPIO
+    _logger: logging.Logger
+
+    def initialize(self, i2c_address: str, gpio_address: str,
+                   gpio_line: int) -> None:
+        logging.basicConfig(level=logging.INFO)
+        self._logger = logging.getLogger(__name__)
+        self.i2c = I2C(i2c_address)
+        """str: /dev/i2c-3 in original"""
+        self.gpio_out_imu_reset = GPIO(gpio_address, gpio_line, "out")
+        """/dev/gpiochip4"""
+
+    def operation_mode_sel(self) -> None:
+        self.write(Register.OPR_MODE, 0x0C)
+        self._logger.info("operation mode set")
+        """write this for NDOF mode: 0x0C"""
```

Comment:
There is a difference between comments and docstrings. Whatever is within triple quotes ``"""`` is a docstring. The purpose of this line is a comment. You should just precede it with a ``#``.